src/appkit.cs

```diff
@@ -4705,27 +4832,82 @@ partial interface NSControl {
 		void EndEditing ([NullAllowed] NSText textObj);
 	}
 
-	[DesignatedDefaultCtor]
-	[BaseType (typeof (NSObject))]
-	interface NSController : NSCoding {
+	[Protocol]
+	interface NSEditorRegistration
+	{
 		[Export ("objectDidBeginEditing:")]
-		void ObjectDidBeginEditing (NSObject editor);
+		void ObjectDidBeginEditing (INSEditor editor);
 
 		[Export ("objectDidEndEditing:")]
-		void ObjectDidEndEditing (NSObject editor);
+		void ObjectDidEndEditing (INSEditor editor);
+	}
+
+	[Category]
+	[BaseType (typeof(NSObject))]
+	interface NSObject_NSEditorRegistration
+	{
+		[Export ("objectDidBeginEditing:")]
+		void ObjectDidBeginEditing (INSEditor editor);
+
+		[Export ("objectDidEndEditing:")]
+		void ObjectDidEndEditing (INSEditor editor);
+	}
+	
+	interface INSEditor {}
 
+	[Protocol]
+	interface NSEditor
+	{
+		[Abstract]
 		[Export ("discardEditing")]
 		void DiscardEditing ();
 
+		[Abstract]
 		[Export ("commitEditing")]
-		bool CommitEditing { get; }
+		bool CommitEditing (); 
```

Comment:
that's likely a good change - but it might bring conflicts...

Reply:
The diff is misleading. I did _not_ break API compat here, as much as I wanted it.

It was originally on NSController and Apple pulled out half of it to the NSEditor protocol. I could fix the obviously wrong thing when I rebound, but it meant that NSController couldn't be an NSEditor.

It's a bit of a mess.